context/app/static/js/components/types.ts

```diff
@@ -0,0 +1,89 @@
+export type DonorEntityType = 'Donor';
+export type SampleEntityType = 'Sample';
+export type DatasetEntityType = 'Dataset';
+export type DataEntityType = DonorEntityType | SampleEntityType | DatasetEntityType;
+export type CellTypeEntityType = 'CellType';
+export type GeneEntityType = 'Gene';
+export type VerifiedUserEntityType = 'VerifiedUser';
+export type ExternalEntityType = CellTypeEntityType | GeneEntityType | VerifiedUserEntityType;
+export type WorkspaceEntityType = 'Workspace';
```

Comment:
I tried to distinguish as follows:
- Entities which are submitted by users (donors, samples, and entities) are `DataEntityType`
- Entities which sourced from an external database (cell types, genes, globus user status) are `ExternalEntityType`
- Entities which are primarily used for internal organization or otherwise require HIVE intervention to create (Support, Collection, and Publication) are `InternalEntityType`

Together, the union of these entity types makes up the icons that are available in the `entityIconMap`; since most of these entities are not relevant in contexts relevant to donors/samples/datasets in particular, having separate type definitions we could refer to for those cases makes it more convenient to use these helper types (i.e. we don't have to `Pick<EntityType, 'Donor'|'Sample'|'Dataset'>` anymore).

I am open to alternative ideas for how we could organize this better.

Reply:
Hmm, I mostly want to distinguish es data vs. other sources.